### PR TITLE
Update a peer with a new packet

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: ntp-proto/src/time_types.rs [hunk 1]
```diff
@@ -179,12 +179,34 @@ impl NtpDuration {
...
+        Self::from_seconds(log2d(input))
+    }
+
+    #[must_use]
+    pub(crate) const fn divided_by(self, divisor: i64) -> Self {
+        // No overflow risks for division
+        Self {
+            duration: self.duration / divisor,
+        }
+    }
```
Comment (folkertdev) [inline ntp-proto/src/time_types.rs:192]: I needed const division.

File: ntp-proto/src/time_types.rs [hunk 2]
```diff
@@ -179,12 +179,34 @@ impl NtpDuration {
         Self { duration }
     }
 
+    pub(crate) fn from_exponent(input: i8) -> Self {
+        Self::from_seconds(log2d(input))
+    }
```
Comment (folkertdev) [inline ntp-proto/src/time_types.rs:184]: so technically the exponents are `i8`s. In practice they are usually `u8`s so we may want a custom function for that (which could be `const`)

File: ntp-proto/src/time_types.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/time_types.rs:192]: I guess this is not fixable by adding const to the div implementation?

Comment (davidv1992) [reply ntp-proto/src/time_types.rs:192]: I guess this is not fixable by adding const to the div implementation?

File: ntp-proto/src/filter.rs [hunk 3]
```diff
@@ -299,6 +328,124 @@ impl Peer {
...
+            let delay = BROADCAST_DELAY;
+            let dispersion =
+                packet_precision + system_precision + multiply_by_phi(BROADCAST_DELAY * 2i64);
+
+            FilterTuple {
+                offset,
+                delay,
+                dispersion,
+                time: local_clock_time,
+            }
```
Comment (davidv1992) [inline ntp-proto/src/filter.rs:384]: Lets not deal with broadcast yet.

File: ntp-proto/src/filter.rs [hunk 4]
```diff
@@ -299,6 +328,124 @@ impl Peer {
...
+                self.host_poll,
+                NtpDuration::from_exponent(self.last_packet.poll),
+            );
+            self.next_date = self.out_date + offset;
+        }
+
+        if self.next_date < local_clock_time {
+            self.next_date = local_clock_time + NtpDuration::ONE;
+        }
+    }
```
Comment (davidv1992) [inline ntp-proto/src/filter.rs:440]: I don't quite see what this is supposed to do?

File: ntp-proto/src/filter.rs (hunk 3 above)
Comment (folkertdev) [reply ntp-proto/src/filter.rs]: finding all the places where we do need to add it later will be a nightmare. When the cost is small, I figure we can just do the implementation. Or leave an explicit `todo!()`

File: ntp-proto/src/time_types.rs (hunk 1 above)
Comment (folkertdev) [reply ntp-proto/src/time_types.rs:193]: sadly not, traits cannot have const methods yet.

**Pull request closed without merging**